phpBB/adm/style/admin.js

```diff
@@ -246,6 +246,57 @@ function parse_document(container)
 	});
 }
 
+async function send_ajax_request(elem)
+{
+	const method = (elem.hasAttribute('data-method')) ? elem.getAttribute('data-method') : 'get';
+	const params = new URLSearchParams();
+	if (method === 'post')
+	{
+		for (const attr of elem.attributes) 
+		{
+			if (attr.name.startsWith('data-post-'))
+			{
+				params.append(attr.name.substr(10).replace('-', '_'), attr.value);
+			}
+		}
+	}
+	
+	const response = await fetch(elem.getAttribute('data-url'), {
+		method: method,
+		body: params,
+	});
+	if (!response.ok)
+	{
+		phpbb.alert(response.status, response.statusText);
+		return;
+	}
+	
+	const resp = await response.json();
+	switch (resp.type)
+	{
+		case 'error':
+			phpbb.alert(resp.status, resp.error);
+			return;
```

Comment:
I guess you don't like break?

Reply:
i have no problem with it, but you may or may not want further code to execute based on the status, hence return seemed more fitting.